File: cpp/1014.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    int maxScoreSightseeingPair(vector<int>& A) {
        int n = A.size();
        int max_score = 0;
        int pre = A[0] + 0, ans = 0;
        for (int j = 1; j < n; j++) {
            ans = max(ans, pre + A[j] - j);
            pre = max(pre, A[j]+j);
        }
        return ans;
    }
};

File: python/1014.py
class Solution:
    def maxScoreSightseeingPair(self, A: List[int]) -> int:
        pre = A[0] + 0
        ans = 0
        for i in range(1, len(A)):
            ans = max(pre + A[i] - i, ans)
            pre = max(A[i] + i, pre)
        return ans